core/src/test/java/org/apache/iceberg/TestMetrics.java

```diff
@@ -766,24 +771,38 @@ protected void assertCounts(int fieldId, Long valueCount, Long nullValueCount, M
 
   protected void assertCounts(
       int fieldId, Long valueCount, Long nullValueCount, Long nanValueCount, Metrics metrics) {
-    Map<Integer, Long> valueCounts = metrics.valueCounts();
-    Map<Integer, Long> nullValueCounts = metrics.nullValueCounts();
-    Map<Integer, Long> nanValueCounts = metrics.nanValueCounts();
-    Assert.assertEquals(valueCount, valueCounts.get(fieldId));
-    Assert.assertEquals(nullValueCount, nullValueCounts.get(fieldId));
-    Assert.assertEquals(nanValueCount, nanValueCounts.get(fieldId));
+    assertThat(metrics.valueCounts())
+        .satisfiesAnyOf(
+            valueCounts -> assertThat(valueCounts).containsEntry(fieldId, valueCount),
+            valueCounts -> assertThat(valueCounts).doesNotContainKey(fieldId));
+
+    assertThat(metrics.nullValueCounts())
+        .satisfiesAnyOf(
+            nullValueCounts -> assertThat(nullValueCounts).containsEntry(fieldId, nullValueCount),
+            nullValueCounts -> assertThat(nullValueCounts).doesNotContainKey(fieldId));
+
+    assertThat(metrics.nanValueCounts())
+        .satisfiesAnyOf(
+            nanValueCounts -> assertThat(nanValueCounts).containsEntry(fieldId, nanValueCount),
+            nanValueCounts -> assertThat(nanValueCounts).doesNotContainKey(fieldId));
   }
 
   protected <T> void assertBounds(
       int fieldId, Type type, T lowerBound, T upperBound, Metrics metrics) {
-    Map<Integer, ByteBuffer> lowerBounds = metrics.lowerBounds();
-    Map<Integer, ByteBuffer> upperBounds = metrics.upperBounds();
-
-    Assert.assertEquals(
-        lowerBound,
-        lowerBounds.containsKey(fieldId) ? fromByteBuffer(type, lowerBounds.get(fieldId)) : null);
-    Assert.assertEquals(
-        upperBound,
-        upperBounds.containsKey(fieldId) ? fromByteBuffer(type, upperBounds.get(fieldId)) : null);
+    assertThat(metrics.lowerBounds())
```

Comment:
this doesn't do the exact same assertion that was done previously. Previously it also asserted that `lowerBound` is `null` when `fieldId` couldn't be found

Reply:
Thanks. Change this part like the previous tests.